webserver/webserver/src/main/java/io/helidon/webserver/BareResponseImpl.java

```diff
@@ -78,19 +78,19 @@
     private volatile DefaultHttpResponse response;
     private volatile boolean lengthOptimization;
     private volatile boolean isWebSocketUpgrade = false;
-    private CompletableFuture<?> previousRequestFuture;
+    private CompletableFuture<?> prevRequestFuture;
```

Comment:
I am unsure of the threading implications of who can modify this field, but I see it being modified in several places.  Does it need to be `volatile`?

Reply:
Reading and writing this variable as a volatile as we do in `onNext` is odd and typically flagged as a problem (non-atomic update). Method `onNext` cannot be called in parallel, but it can be called sequentially in different threads. Perhaps this should be an `AtomicReference` instead.

Reply:
…although `AtomicReference` is really just `volatile` memory semantics on steroids, right?  Up to you; just saw it being assigned in various places and wanted to double-check.

Reply:
`onNext`/`onComplete` from which these are invoked are totally ordered. The inter-thread semantics are ensured by the `Publisher` that calls these.

On the other hand, if you have a non-compliant `Publisher` that may issue `onNext` concurrently, the chaining logic is going to be broken even if `prev` is volatile.